LanguageFeatures/Patterns/variable_A03_t01.dart

```diff
@@ -41,7 +41,7 @@ main() {
 //^
 // [analyzer] unspecified
 // [cfe] unspecified
-  final [c, String d] = [1, "2"];
+  final [int c, String d] = [1, "2"];
```

Comment:
Has it actually been fixed? To me it looks like `[1, "2"]` infers as `<Object>[1, "2"]` and both `int c` and `String d` are variable patterns with a built-in downcast, and they are errors in a declaration context. Cf. [this sentence](https://github.com/dart-lang/language/blob/master/accepted/future-releases/0546-patterns/feature-specification.md#:~:text=Also%20variable%2C%20list%2C%20map%2C%20record%2C%20and%20object%20patterns%20only%20match%20a%20value%20of%20a%20certain%20required%20type.%20These%20patterns%20are%20prohibited%20in%20an%20irrefutable%20context%20if%20the%20matched%20value%20isn%27t%20assignable%20to%20that%20type.).

Reply:
The type schema for list pattern here is the greatest lower bound of `int` and `String`, so `Never`, e.i. we infer `final [int c, String d] = <Never>[1, "2"];`, which is an error.

Reply:
Ah, I got the direction wrong: The pattern actually [gets an inferred type argument](https://github.com/dart-lang/language/blob/master/accepted/future-releases/0546-patterns/feature-specification.md#:~:text=the%20inner%20subpattern.-,List%3A,-A%20context%20type) which is then carried over to the initializing expression. But it's still an error. ;-)

So I have several comments which are wrong in the same way. Thanks for clarifying this!